src/vmm/tests/integration_tests.rs

```diff
@@ -47,48 +48,83 @@ fn test_setup_serial_device() {
-        let mut event_manager = EventManager::new().unwrap();
-        let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();
+            let vmm = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter).unwrap();
 
-        let vmm = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter).unwrap();
-        // This exits the process, so we won't get the output from cargo.
-        vmm.lock().unwrap().stop(0);
+            // This test kernel exits after saying "Hello world". Give the vCPUs some time to exit.
+            thread::sleep(Duration::from_millis(200));
+            vmm.lock().unwrap().stop(0);
+        }
+        vmm_pid => {
+            // Parent process: wait for the vmm to exit.
+            let mut vmm_status: i32 = -1;
+            let pid_done = unsafe { libc::waitpid(vmm_pid, &mut vmm_status, 0) };
+            assert_eq!(pid_done, vmm_pid);
+            restore_stdin();
+            // If any panics occurred, its exit status will be != 0.
+            assert!(unsafe { libc::WIFEXITED(vmm_status) });
+            assert_eq!(unsafe { libc::WEXITSTATUS(vmm_status) }, 0);
+        }
     }
 }
 
 #[test]
 fn test_vmm_seccomp() {
     // Tests the behavior of a customized seccomp filter on the VMM.
-    let boot_source_cfg: BootSourceConfig = MockBootSourceConfig::new().with_boot_args().into();
-    let resources: VmResources = MockVmResources::new()
-        .with_boot_source(boot_source_cfg)
-        .into();
-    let mut event_manager = EventManager::new().unwrap();
-
-    register_signal_handler(libc::SIGSYS, mock_sigsys_handler).unwrap();
-    // The customer "forgot" to whitelist the KVM_RUN ioctl.
-    let filter: BpfProgram = MockSeccomp::new().without_kvm_run().into();
+    let pid = unsafe { libc::fork() };
+    match pid {
+        0 => {
+            // Child process: build vmm and (try to) run it.
+            let boot_source_cfg: BootSourceConfig =
+                MockBootSourceConfig::new().with_default_boot_args().into();
+            let resources: VmResources = MockVmResources::new()
+                .with_boot_source(boot_source_cfg)
+                .into();
+            let mut event_manager = EventManager::new().unwrap();
 
-    let vmm = build_microvm(&resources, &mut event_manager, &filter).unwrap();
-    // Give the signal handler some time to complete.
-    thread::sleep(Duration::from_millis(30));
-    assert!(unsafe { SIGSYS_RECEIVED });
-    // This exits the process, so we won't get the output from cargo.
-    vmm.lock().unwrap().stop(0);
+            // The customer "forgot" to whitelist the KVM_RUN ioctl.
+            let filter: BpfProgram = MockSeccomp::new().without_kvm_run().into();
+            let _ = build_microvm(&resources, &mut event_manager, &filter).unwrap();
+            // Give the vCPUs a chance to attempt KVM_RUN.
+            thread::sleep(Duration::from_millis(200));
+            // Should never get here.
+            loop {}
```

Comment:
```suggestion
            vmm.lock().unwrap().stop(some_invalid_exit_code);
```

otherwise parent `waitpid` might spin forever (or at least rust test timeout period).

Reply:
Done